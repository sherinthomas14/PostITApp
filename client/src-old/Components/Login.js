import {
  Form,
  Input,
  FormGroup,
  Label,
  Container,
  Button,
  Col,
  Row,
} from "reactstrap";

import logo from "../Images/logo-t.png";
import { Link } from "react-router-dom";
import { useState } from "react";
import { useDispatch } from "react-redux";
import { login } from "../Features/UserSlice";
import { useSelector } from "react-redux";
import { useEffect } from "react";
import { useNavigate } from "react-router-dom";

const Login = () => {
  const [email, setemail] = useState();
  const [password, setpassword] = useState();

  const dispatch = useDispatch();
  const navigate = useNavigate();

  //Retrieve the current value of the state from the store, name of state is users with a property user
  const user = useSelector((state) => state.users.user);
  const isSuccess = useSelector((state) => state.users.isSuccess);
  const isError = useSelector((state) => state.users.isError);

  const handleLogin = () => {
    const userData = {
      email: email,
      password: password,
    };
    dispatch(login(userData));
  };

  useEffect(() => {
    if (isError) {
      navigate("/login");
    }
    if (isSuccess) {
      navigate("/");
    }
  }, [user, isError, isSuccess]);

  return (
    <div>
      <Container>
        <Form>
          <Row>
            <Col md={3}>
              <img src={logo} />
            </Col>
          </Row>
          <Row>
            <Col md={3}>
              <FormGroup>
                <Label for="email">Email</Label>
                <Input
                  id="email"
                  name="email"
                  placeholder="Enter email..."
                  type="email"
                  onChange={(e) => setemail(e.target.value)}
                />
              </FormGroup>
            </Col>
          </Row>
          <Row>
            <Col md={3}>
              <FormGroup>
                <Label for="password">Password</Label>
                <Input
                  id="password"
                  name="password"
                  placeholder="Enter password..."
                  type="password"
                  onChange={(e) => setpassword(e.target.value)}
                />
              </FormGroup>
            </Col>
          </Row>
          <Row>
            <Col md={3}>
              <Button onClick={() => handleLogin()}> Login </Button>
            </Col>
          </Row>
        </Form>
        <p className="smalltext">
          No Account? <Link to="/register">Sign Up now.</Link>
        </p>
      </Container>
    </div>
  );
};

export default Login;
