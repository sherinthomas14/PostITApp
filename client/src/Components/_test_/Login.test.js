import renderer from 'react-test-renderer';
import Login from '../Login';
import React from 'react';
import { render, screen, fireEvent } from '@testing-library/react';
import '@testing-library/jest-dom';
import { Provider } from 'react-redux';
import { BrowserRouter as Router } from 'react-router-dom';
import configureStore from 'redux-mock-store';

const mockStore = configureStore([]);
const store = mockStore({
  users: { isSuccess: false, isError: false, user: null },
});

test('matches the UI snapshot with screen.debug()', () => {
    const { container } = render(
      <Provider store={store}>
        <Router>
          <Login />
        </Router>
      </Provider>
    );
    screen.debug(container);
    expect(container).toMatchSnapshot();
  });
  test('validates email format using regex', () => {
    render(
      <Provider store={store}>
        <Router>
          <Login />
        </Router>
      </Provider>
    );
  
    const emailInput = screen.getByLabelText(/email/i);
    fireEvent.change(emailInput, { target: { value: 'valid.email@example.com' } });
  
    const emailRegex = /^[^\s@]+@[^\s@]+\.[^\s@]+$/;
    expect(emailRegex.test(emailInput.value)).toBe(true);
  
    fireEvent.change(emailInput, { target: { value: 'invalid-email' } });
    expect(emailRegex.test(emailInput.value)).toBe(false);
  });
  
  test('validates password format using regex', () => {
    render(
      <Provider store={store}>
        <Router>
          <Login />
        </Router>
      </Provider>
    );
  
    const passwordInput = screen.getByLabelText(/password/i);
    const passwordRegex = /^(?=.*[A-Z])(?=.*\d)(?=.*[!@#$%^&*(),.?":{}|<>]).{6,}$/;
  
    fireEvent.change(passwordInput, { target: { value: 'Abc@123' } });
    expect(passwordRegex.test(passwordInput.value)).toBe(true);
  });
  
  const initval1={
    user:{},
    isSuccess: false,
    isError: false,
    isLoading: false,
}
test("Should return initial state", () => {
    expect(
      reducer(undefined, {
        type: undefined,
      })
    ).toEqual(initval1);
  });


    
  
  