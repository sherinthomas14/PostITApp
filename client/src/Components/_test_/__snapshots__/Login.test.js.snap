// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches the UI snapshot with screen.debug() 1`] = `
<div>
  <div>
    <div
      class="container"
    >
      <form
        class=""
      >
        <div
          class="row"
        >
          <div
            class="col-md-3"
          >
            <img
              src="logo-t.png"
            />
          </div>
        </div>
        <div
          class="row"
        >
          <div
            class="col-md-3"
          >
            <div
              class="mb-3"
            >
              <label
                class="form-label"
                for="email"
              >
                Email
              </label>
              <input
                class="form-control"
                id="email"
                name="email"
                placeholder="Enter email..."
                type="email"
              />
            </div>
          </div>
        </div>
        <div
          class="row"
        >
          <div
            class="col-md-3"
          >
            <div
              class="mb-3"
            >
              <label
                class="form-label"
                for="password"
              >
                Password
              </label>
              <input
                class="form-control"
                id="password"
                name="password"
                placeholder="Enter password..."
                type="password"
              />
            </div>
          </div>
        </div>
        <div
          class="row"
        >
          <div
            class="col-md-3"
          >
            <button
              class="btn btn-secondary"
              type="button"
            >
               Login 
            </button>
          </div>
        </div>
      </form>
      <p
        class="smalltext"
      >
        No Account? 
        <a
          href="/register"
        >
          Sign Up now.
        </a>
      </p>
    </div>
  </div>
</div>
`;
